### PR TITLE
Node Filters

### PR DESCRIPTION
**Pull request by Mahmoud-Emad**
### Description

- Resolved the reset filter issues by implementing a function that returns an object with the default values. This approach replaces the previous direct object usage `() => obj`.
- Reset filters in the farms page, updated the nodes page, and addressed an issue related to the max integer value in the filters.

### Changes

- Update Node Filters and Conversion to Vue 3 Composition API
- Fix Reset Filter Issues using a Function that returns an object instead of using a direct object
- Reset Filters in the Farms Page, Nodes Page Update, and Max Int Value Fix

### Related Issues

- https://github.com/threefoldtech/tfgrid-sdk-ts/issues/1561
- https://github.com/threefoldtech/tfgrid-sdk-ts/issues/1494

### Screenshots/Video

[Screencast from 12-05-2023 03:01:35 PM.webm](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/57001890/eb4b9f3d-092c-472b-8ca5-8baa859a768c)

### Checklist

- [ ] Tests included
- [x] Build pass
- [x] Documentation
- [x] Code format and docstrings
- [x] Screenshots/Video attached (needed for UI changes)

### CODE REVIEW

**Comment by AlaaElattar** [comment]
- The hint in explorer farms for `Farm ID` is different from hint in explorer nodes.
- also it should be `Farm ID` in both of them.

image

image

**Comment by AlaaElattar** [comment]
- How pagination works when there is no data ?
[Screencast from 12-06-2023 11:20:57 AM.webm](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/54944491/d085ca5c-3000-4a42-88bf-6ce16ac8319e)

**Review by mohamedamer453 — APPROVED** [review]

**Comment by AlaaElattar** [comment]
- What do u think about hiding the arrow icon from css style, if we keep it open always and ?
- It gives me a feeling that i can close it but it won't close.

**Comment by AlaaElattar** [comment]
> * How pagination works when there is no data ?
>   [Screencast from 12-06-2023 11:20:57 AM.webm](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/54944491/d085ca5c-3000-4a42-88bf-6ce16ac8319e)

I think it's out of scope of the pr, I will open an issue for it.

File: packages/playground/src/utils/filter_farms.ts [hunk 1]
```diff
@@ -1,42 +1,45 @@
...
   freeIps: InputFilterType;
 };
-export const inputsInitializer: FilterFarmInputs = {
+
+export const inputsInitializer: () => FilterFarmInputs = () => ({
   farmId: {
-    label: "Farm IDs",
-    placeholder: "e.g. 1,2,3.",
+    label: "Farm ID",
+    placeholder: "Filter by farm id",
```
Comment (amiraabouhadid) [inline packages/playground/src/utils/filter_farms.ts:13]: make all placeholders consistent by adding e.g 1,2,3 etc. also capitalize ID

**Comment by Mahmoud-Emad** [comment]
> * The hint in explorer farms for `Farm ID` is different from hint in explorer nodes.
> * also it should be `Farm ID` in both of them.
> 
> image
> 
> image

Hey @AlaaElattar 

the explorer farm (IDs)
the farms farm(ID)

I saw it in the old dashboard it was like I did, and the implementation of it was searched by farm ID since we are in the farm view itself

if you propose that we make all of them similar, we'll need to open an issue for that feature, what do you think?

File: packages/playground/src/views/farms.vue [hunk 2]
```diff
@@ -169,12 +172,17 @@ watch(mixedFarmFilters, updateFarms, { deep: true });
 watch(filterFarmInputs, updateQueries, { deep: true });
 watch(page, updateQueries);
 watch(size, updateQueries);
+
+// The filters should reset to the default value again..
 const inputFiltersReset = (nFltrNptsVal: FilterFarmInputs) => {
```
Comment (amiraabouhadid) [inline packages/playground/src/views/farms.vue:177]: change the variable name nFltrNptsVal to something understandable.

File: yarn.lock [hunk 3]
```diff
@@ -4323,6 +4323,11 @@
   resolved "https://registry.npmjs.org/@types/crypto-js/-/crypto-js-4.1.1.tgz"
   integrity sha512-BG7fQKZ689HIoc5h+6D2Dgq1fABRa0RbBWKBd9SP/MVRVXROflpm5fhwyATX5duFmbStzyzyycPB8qUYKDH3NA==
 
+"@types/crypto-js@^4.2.1":
+  version "4.2.1"
+  resolved "https://registry.yarnpkg.com/@types/crypto-js/-/crypto-js-4.2.1.tgz#480edd76991a63050cb88db1a8840758c55a7135"
+  integrity sha512-FSPGd9+OcSok3RsM0UZ/9fcvMOXJ1ENE/ZbLfOPlBWj7BgXtEAM8VYfTtT760GiLbQIMoVozwVuisjvsVwqYWw==
+
```
[the remark this answers is not in the document]
Comment (Mahmoud-Emad) [reply yarn.lock:4330]: I just did yarn :D

File: packages/playground/src/views/nodes.vue [hunk 4]
```diff
@@ -179,14 +180,17 @@ export default {
       { deep: true },
     );
 
-    // The mixed filters should reset to the default value again..
+    // The filters should reset to the default value again..
     const inputFiltersReset = (nFltrNptsVal: FilterInputs) => {
```
Comment (amiraabouhadid) [inline packages/playground/src/views/nodes.vue:184]: Here too , make nFltrNptsVal filtersInputValues for example

File: packages/playground/src/utils/filter_farms.ts [hunk 5]
```diff
@@ -1,42 +1,45 @@
...
   freeIps: {
     label: "Free PubIPs",
-    placeholder: "e.g. 1,2,3.",
+    placeholder: "Filter by free public ips",
     rules: [
       [
         isNumeric("This field accepts numbers only.", { no_symbols: true }),
         min("Free Public IP should be larger than zero.", 1),
         isInt("should be an integer"),
+        validateResourceMaxNumber("This is not a valid public ip."),
```
Comment (amiraabouhadid) [inline packages/playground/src/utils/filter_farms.ts:40]: should be IP, not ip

File: packages/playground/src/utils/filter_farms.ts [hunk 6]
```diff
@@ -1,42 +1,45 @@
...
-    placeholder: "e.g myfarm",
+    placeholder: "Farm name, e.g myfarm",
     rules: [[isString("Farm name should be made of either numbers or letters")]],
     type: "text",
   },
 
   freeIps: {
     label: "Free PubIPs",
-    placeholder: "e.g. 1,2,3.",
+    placeholder: "Filter by free public ips",
```
Comment (amiraabouhadid) [inline packages/playground/src/utils/filter_farms.ts:34]: IPs not ips

**Review by amiraabouhadid — CHANGES_REQUESTED** [review]

File: packages/playground/src/dashboard/components/dedicated_nodes_table.vue [hunk 7]
```diff
@@ -163,6 +168,12 @@ const tabParams = {
   },
 };
 
+// The mixed filters should reset to the default value again..
+const inputFiltersReset = (nFltrNptsVal: DedicatedNodeFilters) => {
```
Comment (amiraabouhadid) [inline packages/playground/src/dashboard/components/dedicated_nodes_table.vue:172]: here too, change nFltrNptsVal

**Comment by amiraabouhadid** [comment]
Please make the HDD filter in dedicated nodes use TB instead of GB to be consistent with the unit used in the dedicated nodes table 
Screenshot from 2023-12-06 12-09-28

File: packages/playground/src/views/farms.vue (hunk 2 above)
Comment (Mahmoud-Emad) [reply packages/playground/src/views/farms.vue:177]: I renamed it, Please give it a look again :)

**Review by zaelgohary — APPROVED** [review]

**Review by amiraabouhadid — APPROVED** [review]

**Pull request merged**